x/examples/mobileproxy-clib/main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+// #include <stdlib.h>
+import (
+	"C"
+	"runtime/cgo"
+	"unsafe"
+
+	"github.com/Jigsaw-Code/outline-sdk/x/mobileproxy"
+)
+
+type StreamDialerPtr = unsafe.Pointer
+type ProxyPtr = unsafe.Pointer
+
+//export NewStreamDialerFromConfig
+func NewStreamDialerFromConfig(config *C.char) StreamDialerPtr {
+	streamDialerObject, err := mobileproxy.NewStreamDialerFromConfig(C.GoString(config))
+
+	if err != nil {
+		// TODO: print something?
+		return unsafe.Pointer(nil)
+	}
+
+	streamDialerHandle := cgo.NewHandle(streamDialerObject)
+
+	return unsafe.Pointer(&streamDialerHandle)
+}
+
+//export RunProxy
+func RunProxy(address *C.char, dialer StreamDialerPtr) ProxyPtr {
+	dialerObject := (*cgo.Handle)(dialer).Value().(mobileproxy.StreamDialer)
+
+	proxyObject, err := mobileproxy.RunProxy(C.GoString(address), &dialerObject)
+
+	if err != nil {
+		// TODO: print something?
```

Comment:
We must have a way to return errors. Go Mobile returns a struct instead.

Reply:
Yes, you can try to return a tuple and paste the header here.

@fortuna, do you mean a more C-friendly way of returning errors, like below?

```
Proxy *proxy;
int err = CreateProxy(&proxy);
```

Reply:
We need a way to return an error meesage.

Also, which design should we use?

- Return error (proxy reference)
- Return proxy (error reference)
- Return tuple
- return nothing (error and proxy reference)

I think Go Mobile on ios returns the value and writes the error to an input reference.

Reply:
`errno` is very Unix specific. But sure, we can introduce our own `errno`. That's actually my code above, we'll return an `int`, and put the proxy into a "output parameter".

Reply:
We need a way to return an error message too. Errno is not enough.